zk/hermez_db/db_xlayer.go

```diff
@@ -59,3 +66,51 @@ func (db *HermezDbReader) GetInnerTxs(blockNum uint64) [][]*types.InnerTx {
 	}
 	return blockInnerTxs
 }
+
+// TruncateInnerTx deletes all inner txs of a block
+func (db *HermezDb) TruncateInnerTx(block uint64) error {
+	prefix := make([]byte, 8)
+	binary.BigEndian.PutUint64(prefix, block)
+
+	it, err := db.tx.Prefix(INNER_TX, prefix)
+	if err != nil {
+		log.Error("inner txs fetching failed", "err", err)
+		return nil
+	}
+	defer func() {
+		if casted, ok := it.(kv.Closer); ok {
+			casted.Close()
+		}
+	}()
+
+	var keyList [][]byte
+	for it.HasNext() {
+		k, v, err := it.Next()
+		if err != nil {
+			log.Error("inner txs fetching failed", "err", err)
+			return nil
+		}
+		innerTxs := make([]*types.InnerTx, 0)
+		if err := rlp.DecodeBytes(v, &innerTxs); err != nil {
```

Comment:
I noticed you decoded the value bytes into `innerTxs` variable. But is this variable used afterwards? I am wondering if we could skip the `make` and discard the decoded variable on the stack immediately after decoding?

```suggestion
		if err := rlp.DecodeBytes(v, []*types.InnerTx{}); err != nil {
```

Reply:
Deleted this decode code.